packages/insured-bridge-relayer/src/ProfitabilityCalculator.ts

```diff
@@ -0,0 +1,227 @@
+
+    // Calculate the expected revenue, in ETH, based on the amount of tokens being offered in fees.
+    const ethRevenue = this.getEthRevenue(tokenEthPrice, slowRevenue, speedUpRevenue, instantRevenue);
+
+    // Calculate the relay profitability as the difference between revenue and cost.
+    const ethProfitability = this.getProfit(ethSubmissionCost, ethRevenue);
+
+    // Finally, determine the relay submit type based on the profitability.
+    let relaySubmitType: RelaySubmitType;
+    if (
+      ethProfitability.instantEthProfit.gt(toBN(0)) &&
+      ethProfitability.instantEthProfit.gt(ethProfitability.speedUpEthProfit) &&
+      ethProfitability.instantEthProfit.gt(ethProfitability.slowEthProfit)
+    )
+      relaySubmitType = RelaySubmitType.Instant;
+    else if (
+      ethProfitability.speedUpEthProfit.gt(toBN(0)) &&
+      ethProfitability.speedUpEthProfit.gt(ethProfitability.slowEthProfit)
+    )
+      relaySubmitType = RelaySubmitType.SpeedUp;
+    else if (ethProfitability.slowEthProfit.gt(toBN(0))) relaySubmitType = RelaySubmitType.Slow;
+    else relaySubmitType = RelaySubmitType.Ignore;
+
+    this.logger.debug({
+      at: "ProfitabilityCalculator",
+      message: "Considered relay profitability",
+      l1Token,
+      tokenType: TokenType[tokenType],
+      tokenEthPrice: fromWei(tokenEthPrice),
+      cumulativeGasPrice: cumulativeGasPrice.toString(),
+      relayerDiscount: fromWei(this.relayerDiscount),
+      ethSubmissionCost: objectMap(ethSubmissionCost, (value: BN) => fromWei(value)),
+      ethRevenue: objectMap(ethRevenue, (value: BN) => fromWei(value)),
+      ethProfitability: objectMap(ethProfitability, (value: BN) => fromWei(value)),
+      relaySubmitType: RelaySubmitType[relaySubmitType],
+    });
+    return relaySubmitType;
+  }
+
+  getRelayEthSubmissionCost(
+    gasPrice: BN,
+    tokenType: TokenType
+  ): {
+    slowEThCost: BN;
+    speedUpEthCost: BN;
+    instantEthCost: BN;
+  } {
+    const discount = fixedPoint.sub(this.relayerDiscount);
+    switch (tokenType) {
+      case TokenType.WETH:
+        return {
+          slowEThCost: gasPrice.mul(toBN(across.constants.SLOW_ETH_GAS)).mul(discount).div(fixedPoint),
+          speedUpEthCost: gasPrice.mul(toBN(across.constants.SPEED_UP_ETH_GAS)).mul(discount).div(fixedPoint),
+          instantEthCost: gasPrice.mul(toBN(across.constants.FAST_ETH_GAS)).mul(discount).div(fixedPoint),
+        };
+
+      case TokenType.ERC20:
+        return {
+          slowEThCost: gasPrice.mul(toBN(across.constants.SLOW_ERC_GAS)).mul(discount).div(fixedPoint),
+          speedUpEthCost: gasPrice.mul(toBN(across.constants.SPEED_UP_ERC_GAS)).mul(discount).div(fixedPoint),
+          instantEthCost: gasPrice.mul(toBN(across.constants.FAST_ERC_GAS)).mul(discount).div(fixedPoint),
+        };
+
+      case TokenType.UMA:
+        return {
+          slowEThCost: gasPrice.mul(toBN(across.constants.SLOW_UMA_GAS)).mul(discount).div(fixedPoint),
```

Comment:
Does UMA transfer really have that much of a different gas cost?

Reply:
Also, I've talked to @daywiss about this in a previous PR but we shouldn't have to hardcode these gas costs, can't we use some sort of `contract.methods.estimateGas` to get rid of the `enum TokenType`?

Reply:
i think its good to have some default values, especially for frontend where its better to have the option to get a syncronous value, but if its easy to do i think it would be great to additionally add a way to get this cost dynamically

Reply:
Agreed